frontend/javascripts/oxalis/controller/combinations/tool_controls.ts

```diff
@@ -617,10 +623,62 @@ export class BoundingBoxTool {
     getSceneController().highlightUserBoundingBox(null);
   }
 }
+export class ProofreadTool {
+  static getPlaneMouseControls(_planeId: OrthoView, planeView: PlaneView): any {
+    return {
+      leftClick: (pos: Point2, plane: OrthoView, event: MouseEvent, isTouch: boolean) => {
+        this.onLeftClick(planeView, pos, plane, event, isTouch);
+      },
+    };
+  }
+
+  static onLeftClick(
+    planeView: PlaneView,
+    pos: Point2,
+    plane: OrthoView,
+    _event: MouseEvent,
+    isTouch: boolean,
+  ) {
+    const didSelectNode = SkeletonHandlers.handleSelectNode(planeView, pos, plane, isTouch);
+
+    let globalPosition;
+    if (plane === OrthoViews.TDView) {
+      if (didSelectNode) {
+        getSkeletonTracing(Store.getState().tracing).map((skeletonTracing: SkeletonTracing) =>
+          getNodeAndTree(skeletonTracing).map(([_activeTree, activeNode]) => {
+            globalPosition = activeNode.position;
+          }),
+        );
+      }
+    } else {
+      globalPosition = calculateGlobalPos(Store.getState(), pos);
```

Comment:
The `pos` argument to `calculateGlobalPos` is the _click position_ in 2d viewport coordinates. `calculateGlobalPos` converts this to the global position in 3d. This works great for the orthogonal viewports (and is used for other tools and the context menu). However, this doesn't work for the 3D viewport, because the click position in 3D cannot be determined uniquely. If the user clicked _on a node_ in the 3d viewport, though, we can determine the click position by looking up the node position. That's what's happening here.
I'll add a comment to make this more clear!